extensions/wikia/LookupContribs/SpecialLookupContribs_ajax.php

```diff
@@ -12,159 +13,144 @@
-							$rows[] = array(
-								$loop + $offset, // id
-								$link, // title
-								$diff . ' ' . $hist . ' ' . $contrib, // links to diff, history and user contribution (link to special page)
-								$edit
-							);
-							$loop++;
-						}
+			}
+		} else {
+			$oLC->setDBname( $dbname );
+			$oLC->setMode( $mode );
+			$oLC->setNamespaces( $nspace );
+			$oLC->setLimit( $limit );
+			$oLC->setOffset( $offset );
+			$data = $oLC->fetchContribs();
+
+			$result = [];
+
+			if ( !empty( $data ) && is_array( $data ) ) {
+				$result['iTotalRecords'] = intval( $limit );
+				$result['iTotalDisplayRecords'] = intval( $data['cnt'] );
+				$result['sColumns'] = 'id,title,links,edit';
+				$rows = [];
+				if ( isset( $data['data'] ) ) {
+					$loop = 1;
+					foreach ( $data['data'] as $id => $row ) {
+						list ( $link, $diff, $hist, $contrib, $edit, $removed ) = array_values( $oLC->produceLine( $row ) );
+						$rows[] = [
+							$loop + $offset, // id
+							$link, // title
+							$diff . ' ' . $hist . ' ' . $contrib, // links to diff, history and user contribution (link to special page)
+							$edit
+						];
+						$loop++;
 					}
-					$result['aaData'] = $rows;
 				}
+				$result['aaData'] = $rows;
 			}
 		}
 
-		wfProfileOut( __METHOD__ );
-		return json_encode($result);
+		return json_encode( $result );
 	}
 
 	/**
-	 * @brief Generates row data for user if ajax call was sent from Special:LookupContribs
+	 * Generates row data for user if ajax call was sent from Special:LookupContribs
 	 *
 	 * @param array $activityData data retrieved from LookupContribsCore::checkUserActivity()
 	 *
 	 * @return array
-	 *
-	 * @author Andrzej 'nAndy' Łukaszewski
 	 */
-	public static function prepareLookupContribsData($activityData) {
-		global $wgLang;
-
-		$rows = array();
-		foreach( $activityData as $row ) {
-			$rows[] = array(
+	public static function prepareLookupContribsData( $activityData ) {
+		$rows = [];
+		foreach ( $activityData as $row ) {
+			$rows[] = [
 				$row['id'], // wiki Id
 				$row['dbname'], // wiki dbname
-				$row['title'], //wiki title
+				$row['title'], // wiki title
 				$row['url'], // wiki url
-				$wgLang->timeanddate( wfTimestamp( TS_MW, $row['last_edit'] ), true ), //last edited
-				'' //options
-			);
+				F::app()->wg->Lang->timeanddate( wfTimestamp( TS_MW, $row['last_edit'] ), true ), // last edited
+				'' // options
+			];
 		}
 
 		return $rows;
 	}
 
 	/**
-	 * @brief Generates row data for user if ajax call was sent from Special:LookupUser
+	 * Generates row data for user if ajax call was sent from Special:LookupUser
 	 *
 	 * @param array $activityData data retrieved from LookupContribsCore::checkUserActivity()
 	 *
 	 * @return array
-	 *
-	 * @author Andrzej 'nAndy' Łukaszewski
 	 */
-	public static function prepareLookupUserData($activityData, $username) {
-		global $wgLang, $wgContLang;
+	public static function prepareLookupUserData( $activityData, $username ) {
+		$wg = F::app()->wg;
```

Comment:
I do not think we have a strict standard about it. I adjust my style (or lack thereof) to be consistent within the scope of a particular file or piece of code. Generally, `global` is the way to go with the core code, whereas `F::app()` is for everything Wikia. Also, local scope is always faster than the same operation in global scope. And just declaring a variable as global with `global` (without even using it later) slows things down as PHP needs to check whether it exists and create a reference. Note, that passing by reference in PHP is slower than passing by value. I am fairly sure `F::app()` addresses the problem.